generator/src/main/java/com/scottlogic/deg/generator/fieldspecs/FieldSpecFactory.java

```diff
@@ -272,42 +256,19 @@ private FieldSpec construct(FormatConstraint constraint, boolean negate, boolean
     }
 
     private FieldSpec construct(StringHasLengthConstraint constraint, boolean negate, boolean violated) {
-        final Pattern regex = Pattern.compile(String.format(".{%s}", constraint.referenceValue));
-        return constructPattern(regex, negate, true, constraint, violated);
+        return construct(negate, constraint, violated);
     }
 
     private FieldSpec construct(IsStringShorterThanConstraint constraint, boolean negate, boolean violated) {
-        final Pattern regex = Pattern.compile(String.format(".{0,%d}", constraint.referenceValue - 1));
-        return constructPattern(regex, negate, true, constraint, violated);
+        return construct(negate, constraint, violated);
     }
 
     private FieldSpec construct(IsStringLongerThanConstraint constraint, boolean negate, boolean violated) {
-        final Pattern regex = Pattern.compile(String.format(".{%d,}", constraint.referenceValue + 1));
-        return constructPattern(regex, negate, true, constraint, violated);
+        return construct(negate, constraint, violated);
     }
 
-    private FieldSpec constructPattern(Pattern pattern, boolean negate, boolean matchFullString, AtomicConstraint constraint, boolean violated) {
-        AtomicConstraintConstructTuple atomicConstraintConstructTuple = new AtomicConstraintConstructTuple(constraint, matchFullString, negate);
-        StringGenerator regexStringGenerator;
-        if (constraintToStringGeneratorMap.containsKey(atomicConstraintConstructTuple)) {
-            regexStringGenerator = constraintToStringGeneratorMap.get(atomicConstraintConstructTuple);
-        } else {
-            regexStringGenerator = new RegexStringGenerator(pattern.toString(), matchFullString);
-            if (negate) {
-                regexStringGenerator = regexStringGenerator.complement();
-            }
-            constraintToStringGeneratorMap.put(atomicConstraintConstructTuple, regexStringGenerator);
-        }
-        return construct(regexStringGenerator, negate, constraint, violated);
-    }
-
-    private FieldSpec construct(StringGenerator generator, boolean negate, AtomicConstraint constraint, boolean violated) {
-        final StringRestrictions stringRestrictions = new StringRestrictions(
-            new StringConstraintsCollection(negate
-                ? constraint.negate()
-                : constraint));
-
-        stringRestrictions.stringGenerator = generator;
+    private FieldSpec construct(boolean negate, AtomicConstraint constraint, boolean violated) {
```

Comment:
This is specific to constructing string restrictions, but nothing in its parameters shows this.

Could we name it something like 

constructString(AtomicConstraint constraint, boolean negate, boolean violated)

instead

Reply:
A larger scope change, would put making all of the string constraints into a 
StringConstraint
 interface, that extends
 AtomicConstraint

Reply:
Renamed the overload